plugins/woocommerce/tests/e2e-pw/tests/merchant/product-reviews.spec.js

```diff
@@ -196,6 +199,112 @@ test.describe(
 			).toBeVisible();
 		} );
 
+		test( 'can approve a product review', async ( { page, reviews } ) => {
+			const review = reviews[ 0 ]; // Select the first review for approval
+
+			// Go to the Product Reviews page
+			await page.goto(
+				`wp-admin/edit.php?post_type=product&page=product-reviews`
+			);
+
+			// Locate the review to be approved
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+
+			// Ensure the review is not already approved
+			const approveButton = reviewRow.getByRole( 'button', {
+				name: 'Approve',
+			} );
+
+			// Hover to reveal action buttons
+			await reviewRow.hover();
+
+			// Click the "Approve" button and wait for the state to become 'networkidle'
+			await approveButton.click();
+
+			// Verify the review is approved by checking the presence of "Unapprove" button
+			const unapproveButton = reviewRow.getByRole( 'button', {
+				name: 'Unapprove',
+			} );
+			await expect( unapproveButton ).toBeVisible();
+		} );
+
+		test( 'can mark a product review as spam', async ( {
+			page,
+			reviews,
+		} ) => {
+			const review = reviews[ 0 ]; // Select the first review to mark as spam
+
+			// Go to the Product Reviews page
+			await page.goto(
+				`wp-admin/edit.php?post_type=product&page=product-reviews`
+			);
+
+			// Locate the review to be marked as spam
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+			await reviewRow.hover();
+
+			// Select Spam action and verify the review is marked as spam
+			await reviewRow.getByRole( 'button', { name: 'Spam' } ).click();
+
+			// Navigate to the "Spam" tab to verify the review is marked as spam
+			await page.click( 'a[href*="comment_status=spam"]' );
+
+			// Check if the spammed review is located on the "Spam" page
+			await expect(
+				page.locator( `#comment-${ review.id }` )
+			).toBeVisible();
+		} );
+
+		test( 'can reply to a product review', async ( { page, reviews } ) => {
+			const review = reviews[ 0 ]; // Select the first review to reply to
+
+			// Go to the Product Reviews page
+			await page.goto(
+				'wp-admin/edit.php?post_type=product&page=product-reviews'
+			);
+
+			// Locate the review to be replied to
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+			await reviewRow.hover();
+
+			// Click the Reply button within the review row
+			await reviewRow.getByRole( 'button', { name: 'Reply' } ).click();
+
+			// Wait for the reply textarea to be visible
+			const replyTextArea = page.locator( 'textarea#replycontent' );
+			await expect( replyTextArea ).toBeVisible();
+
+			// Fill in the reply and submit it
+			const replyText = 'Thank you for your feedback!';
```

Comment:
Not super important in here because we're creating the product with each test, but having a unique `replyText` would harden the test. Appending Date.now() should help.

Reply:
Resolved with [c031d68fed14828d3e64eed8a04e505e4c5c88f3](https://github.com/woocommerce/woocommerce/commit/c031d68fed14828d3e64eed8a04e505e4c5c88f3)